### ISSUE TITLE
[a11y]: Unable to have an interactive element with role="button" inside Toggletip

### ISSUE THREAD

**Issue by shreevidya-ibm** [carbon-design-system/carbon#18499]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.62.1

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - latest deployment

### Description

There is an accessibility issue when attempting to use an interactive element, such as a button with role="button", inside the Toggletip component. The Toggletip implementation does not not seem to support or accommodate interactive child elements.

Recommended fix:
Interactive elements (e.g., < Button role="button" >) should be allowed and functional inside the Toggletip component.

### Reproduction/example

https://stackblitz.com/edit/github-qktfqi5o?file=src%2FApp.jsx

### Steps to reproduce

- Open the Target Web Page
- Open Developer Tools
- Navigate to the Accessibility Assessment Tab
- Run the IBM Equal Access Accessibility Checker
- Analyse the Violation stating 'The element with role "button" contains descendants with roles "button" which are ignored by browsers'

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Hi there! You'll need to use Toggletip with it's intended children components as specified in the [documentation](https://react.carbondesignsystem.com/?path=/docs/components-toggletip--overview):

> To use this component, you'll need to include a Toggletip, ToggletipButton for the trigger, and ToggletipContent for the content you would like to render within the popover.

[Here's an example](https://stackblitz.com/edit/github-em3ezcjt?file=src%2FApp.jsx)

If you're looking to use a raw interactive element like a `<button>`, take a look at [Popover](https://react.carbondesignsystem.com/?path=/story/components-popover--default). 

Hope this helps!
